src/tensorlake/documentai/models/options.py

```diff
@@ -0,0 +1,146 @@
+from typing import List, Optional, Type, Union
+
+from pydantic import BaseModel, Field, Json
+
+from .enums import (
+    ChunkingStrategy,
+    FormDetectionMode,
+    ModelProvider,
+    TableOutputMode,
+    TableParsingFormat,
+)
+
+
+class EnrichmentOptions(BaseModel):
+    """
+    Options for enriching a document with additional information.
+
+    This object helps to extend the output of the document parsing process with additional information.
+    This includes summarization of tables and figures, which can help to provide a more comprehensive understanding of the document.
+    """
+
+    figure_summarization: bool = Field(
+        False,
+        description="Boolean flag to enable figure summarization. The default is `false`.",
+    )
+    figure_summarization_prompt: Optional[str] = Field(
+        None,
+        description="The prompt to guide the figure summarization. If not provided, a default prompt will be used. It is not required to provide a prompt. The prompt only has effect if `figure_summarization` is set to `true`.",
+    )
+    table_summarization: bool = Field(
+        False,
+        description="Boolean flag to enable summary generation for parsed tables. The default is `false`.",
+    )
+    table_summarization_prompt: Optional[str] = Field(
+        None,
+        description="The prompt to guide the table summarization. If not provided, a default prompt will be used. It is not required to provide a prompt. The prompt only has effect if `table_summarization` is set to `true`.",
+    )
+
+
+class PageClassConfig(BaseModel):
+    """
+    Configuration for page classification.
+    """
+
+    name: str = Field(description="The name of the page class.")
+    description: str = Field(
+        description="The description of the page class to guide the model to classify the pages. Describe what the model should look for in the page to classify it."
+    )
+
+
+class ParsingOptions(BaseModel):
+    """
+    Options for parsing a document.
+    """
+
+    chunking_strategy: Optional[ChunkingStrategy] = Field(
+        ChunkingStrategy.NONE,
+        description="The chunking strategy determines how the document is chunked into smaller pieces. Different strategies can be used to optimize the parsing process. Choose the one that best fits your use case. The default is `None`, which means no chunking is applied.",
+    )
+    disable_layout_detection: bool = Field(
+        False,
+        description="Useful flag for documents with a lot of tables or images. If set to `true`, the API will skip the layout detection step, and directly extract text from the document.",
+    )
+    form_detection_mode: FormDetectionMode = Field(
+        FormDetectionMode.VLM,
+        description="Algorithm to use for detecting forms in a document.",
+    )
+    remove_strikethrough_lines: bool = Field(
```

Comment:
Didn't we decide to have this be `remove_strikethrough`?

```suggestion
    remove_strikethrough: bool = Field(
```